### PR TITLE
chore: repository cleanup

### PR DESCRIPTION
**Pull request by CalMacCQ**
Cleaning up the repository removing some out of date files and adding some instructions for the new docs build.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,15 +0,0 @@
-# Description
-
-Please summarise the changes.
-
-# Related issues
-
-Please mention any github issues addressed by this PR.
-
-# Checklist
```
Comment (CalMacCQ) [inline .github/pull_request_template.md:9]: I removed this as I think that the checklist is not very helpful for this repository. Maybe we can make a new one.

Comment (cqc-melf) [reply .github/pull_request_template.md:9]: Maybe we can still keep the first two items and just remove the checklist?

Comment (CalMacCQ) [reply .github/pull_request_template.md:9]: Good idea, done.

Comment (CalMacCQ) [reply .github/pull_request_template.md:9]: oops, messed up the commit. Now done in [48f8627](https://github.com/CQCL/pytket-docs/pull/355/commits/48f862764917eb7876b5c380a90782b0b31b8b2b)

**Review by cqc-melf — DISMISSED** [review]

**Pull request merged**